### DISCUSSION TITLE
Hedgehog 0.11.0 feature set

### DISCUSSION THREAD

**Discussion by ghost** [hedgehogqa/fsharp-hedgehog#340] [General]
So, version `0.11.0` has been stagnating for a while. I'd like to redefine the scope of this version so we can get the things we've worked on out the door. Does anyone have any specific work they'd like to see in the next release?

https://github.com/hedgehogqa/fsharp-hedgehog/milestone/2

/cc @moodmosaic @TysonMN @dharmaturtle

**Comment by moodmosaic** [comment]
No specific items from my side 👍

**Comment by dharmaturtle** [comment]
The milestone for ease of reference: https://github.com/hedgehogqa/fsharp-hedgehog/milestone/2

I don't feel strongly about these tickets.

---

On a separate note, I'm building a `TimeboxedPropertyAttribute` for Hedgehog.Xunit. Its documentation:

> Runs tests until the time box is reached or the `tests` parameter is hit (100 by default). If no `size` is provided, a random `size` will be used for each test run.

Raison d'être: I have some tests that take a while to run. During development, I don't care _that_ much about correctness and will therefore timebox said tests. I'll flip them back to normal `property` tests before committing/merging. Since a test might only run once if the timebox is extremely short, I want a random size. (Always testing with `Size=0` isn't great.) This is pretty niche to my workflow, but LMK if any of this sounds interesting and anyone wants it merged upstream.

**Reply by ghost** [reply]
I could definitely see that being useful. From a design point of view, we've typically avoided attributes/reflection in Hedgehog proper (The work in #324 not withstanding, though #326 is meant to address that) and I'd prefer we keep it that way. One of the strengths of Hedgehog to me is the compile-time guarantees it provides over FsCheck.

Are you thinking this could be implemented with another combinator? If so, would this be a property level thing? Or could this be supported with a property config option?

Edit: Also, thanks for referencing the milestone. I've updated the original post to include it.

**Reply by dharmaturtle** [reply]
Opened an issue: https://github.com/hedgehogqa/fsharp-hedgehog/issues/341

**Comment by TysonMN** [comment]
I want to see three things in 0.11:
1. Fix for bug #327, which is just a click away since PR #328 has a fix and is ready to be merged.
2. Recheck only shrunken input (instead of all inputs in the "shrink path" to that input), which is issue #332.  I am actively working on this (see draft PR #336) and think I will be at least somewhat successful.
3. Add applicative behavior, which is #268.  I haven't worked on this yet, but it seems rather straight forward in my mental model.

I am excited to begin working on applicative behavior after I "complete" work on optimized rechecking (for some meaning of "complete" that is yet to be determined).  All my motivation is directed towards Hedgehog lately, and I am working on it almost every evening, so I expect it won't take me too long to complete those last two items.

**Reply by ghost** [reply]
I'll take a look at #328.

Thank you for all of the work you've put into Hedgehog, you've been more-or-less carrying the project recently! 😅 I expect to begin ramping back up on this project soon, so hopefully we can get more features knocked out 👍🏼

**Comment by ghost** [comment]
I've reworked the scope of `0.11.0`. I'm going to prioritize these items so we can get the next version out :+1:

**Reply by ghost** [reply]
I've deferred the remaining `0.11.0` issues and PRs to `0.12.0`.

@moodmosaic I will need to get with you to document the release process.

**Reply by moodmosaic** [reply]
It ought to be all automated now, right? 👀 🍻

Happy to help, of course! Perhaps the only thing we aren't automating is updating our README file as in https://github.com/hedgehogqa/fsharp-hedgehog/pull/304, for example.

**Reply by ghost** [reply]
😄 Speaking of updating the README.. maybe we should throw this in the README or in a wiki page? I know Hedgehog experimental has a small release process section of their README.

**Reply by moodmosaic** [reply]
Oh, sorry, I meant updating the ChangeLog file.
